src/getMedicalCodes.js

```diff
@@ -0,0 +1,18 @@
+function getMedicalCodes(searchTerm, callback) {
+    var rxtermsApiUrl = `https://clinicaltables.nlm.nih.gov/api/rxterms/v3/search?terms=${searchTerm}&ef=STRENGTHS_AND_FORMS`;
+    var icd10ApiUrl = `https://clinicaltables.nlm.nih.gov/api/icd10cm/v3/search?sf=code,name&terms=${searchTerm}`;
+    var loincApiUrl = `https://clinicaltables.nlm.nih.gov/api/loinc_items/v3/search?terms=${searchTerm}&type=panel&df=text,LOINC_NUM`;
+
+    $.get(rxtermsApiUrl, function (rxtermsData) {
+        $.get(icd10cmApiUrl, function (icd10cmData) {
+            $.get(loincApiUrl, function (loincData) {
+                var result = {

```

Comment:
Ideally the caller of this function should not need to know to check each separate code listing; could we instead have a structure like this:

```js
let result = [
  {
    id: “rx/123”,
    name: “acetaminophen”,
    fields: {
      “dose”: “number”,
      “route”: [“oral”, “iv”, “nasogastric”],
      “frequency”: “string”,
      “refills”: “number”
    }
  },
  {
    id: “loinc/453”,
    name: “XR CHEST PA/LAT”,
    fields: {
      priority: [“routine”, “stat”],
      instructions: “string”,
      reason: “string”
    }
  },
  {
    id: “icd/4387”,
    …
  },
  … // more entries follow
]
```

Reply:
I can get this formatting working for RxTerms. LOINC will take longer since it's a smattering of forms, panels, and question codes.